src/Stratis.Bitcoin.IntegrationTests/Wallet/SendingTxWithDoubleSpendSpecification_Steps.cs

```diff
@@ -0,0 +1,104 @@
+            mnemonic1.Words.Length.Should().Equals(12);
+            mnemonic2.Words.Length.Should().Equals(12);
+
+            var addr = this.stratisSender.FullNode.WalletManager().GetUnusedAddress(new WalletAccountReference("mywallet", "account 0"));
+            var wallet = this.stratisSender.FullNode.WalletManager().GetWalletByName("mywallet");
+            var key = wallet.GetExtendedPrivateKeyForAddress("123456", addr).PrivateKey;
+
+            this.stratisSender.SetDummyMinerSecret(new BitcoinSecret(key, this.stratisSender.FullNode.Network));
+            var maturity = (int)this.stratisSender.FullNode.Network.Consensus.Option<PowConsensusOptions>().CoinbaseMaturity;
+
+            this.stratisSender.GenerateStratis(maturity + 5);
+
+            TestHelper.WaitLoop(() => TestHelper.IsNodeSynced(this.stratisSender));
+
+            var total = this.stratisSender.FullNode.WalletManager().GetSpendableTransactionsInWallet("mywallet").Sum(s => s.Transaction.Amount);
+            total.Should().Equals(Money.COIN * 105 * 50);
+ 
+            // sync both nodes
+            this.stratisSender.CreateRPCClient().AddNode(stratisReceiver.Endpoint, true);
+            TestHelper.WaitLoop(() => TestHelper.AreNodesSynced(stratisReceiver, stratisSender));
+        }
+
+        private void coins_first_sent_to_receiving_wallet()
+        {
+            this.sendto = this.stratisReceiver.FullNode.WalletManager().GetUnusedAddress(new WalletAccountReference("mywallet", "account 0"));
+
+            this.transaction = this.stratisSender.FullNode.WalletTransactionHandler().BuildTransaction(WalletTests.CreateContext(
+                new WalletAccountReference("mywallet", "account 0"), "123456", this.sendto.ScriptPubKey, Money.COIN * 100, FeeType.Medium, 101));
+
+            this.stratisSender.FullNode.NodeService<WalletController>().SendTransaction(new SendTransactionRequest(this.transaction.ToHex()));
+
+            TestHelper.WaitLoop(() => this.stratisReceiver.CreateRPCClient().GetRawMempool().Length > 0);
+            TestHelper.WaitLoop(() => this.stratisReceiver.FullNode.WalletManager().GetSpendableTransactionsInWallet("mywallet").Any());
+
+            var receivetotal = this.stratisReceiver.FullNode.WalletManager().GetSpendableTransactionsInWallet("mywallet").Sum(s => s.Transaction.Amount);
+            receivetotal.Should().Equals(Money.COIN * 100);
+            this.stratisReceiver.FullNode.WalletManager().GetSpendableTransactionsInWallet("mywallet").First().Transaction.BlockHeight.Should().BeNull();
+        }
+
+        private void attempt_made_to_spend_same_coins()
+        {
+            //create double spend transaction
+            var sendtoDoubleSpend = this.stratisReceiver.FullNode.WalletManager().GetUnusedAddress(new WalletAccountReference("mywallet", "account 0"));
+            sendtoDoubleSpend.Should().NotBeSameAs(this.sendto);
```

Comment:
I would guess this line could never fail so not needed?

Reply:
Agreed - removed